tern/analyze/common.py

```diff
@@ -563,9 +563,54 @@ def get_git_sha(path_to_toplevel):
     return sha_info
 
 
+def get_git_url(dockerfile_path):
+    '''Given a dockerfile_path, return url of git project which contains
+    the dockerfile in a form of list.'''
+    # get the path of the folder containing the dockerfile
+    dockerfile_folder_path = os.path.dirname(dockerfile_path)
+    command = ['git', 'remote', '-v']
+    try:
+        output = subprocess.check_output(  # nosec
+            command, stderr=subprocess.DEVNULL, cwd=dockerfile_folder_path)
+        if isinstance(output, bytes):
+            lines = output.decode('utf-8').split('\n')
+            # pop the last line which is an empty line
+            lines.pop()
+            url_list = []
+            for line in lines:
+                extract_url = extract_git_url_from_line(line)
+                if extract_url and extract_url not in url_list:
+                    url_list.append(extract_url)
```

Comment:
If `url_list = set()` then you don't have to do the `extract_url not in url_list` check and just do `url_list.add(extract_url)`.